### PR TITLE
NPE fix

### PR DESCRIPTION
**Pull request by lumber1000**
NullPointer exception fixed

### CODE REVIEW

**Review by OptimumCode — APPROVED** [review]
Could you please add a description to README about what is the problem you fixed? Also, it would be great to have tests added for this scenario that caused NPE.
In general, everything looks good to me. But I would prefer if @Nikita-Smirnov-Exactpro looks at the PR before you merge it

File: cradle-cassandra/src/test/java/com/exactpro/cradle/cassandra/dao/messages/MessagesIteratorProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+import java.time.Instant;
+import java.util.Collections;
+
+public class MessagesIteratorProviderTest extends TestCase {
+    protected static final BookId DEFAULT_BOOK_ID = new BookId("test_book");
+    private static final String FIRST_SESSION_ALIAS = "test_session_alias";
+
+    @Test(description = "Create iterator provider with empty book")
+    public void createIteratorProviderWithEmptyBook() throws CradleStorageException {
```
Comment (OptimumCode) [inline cradle-cassandra/src/test/java/com/exactpro/cradle/cassandra/dao/messages/MessagesIteratorProviderTest.java:37]: What do we test here? Test does not have a single assertion

File: cradle-cassandra/build.gradle [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ext {
-    driver_version = '4.15.0'
+    driver_version = '4.17.0'
```
[the remark this answers is not in the document]
Comment (OptimumCode) [reply cradle-cassandra/build.gradle:2]: If cradle uses the BOM then probably yes, we could

File: cradle-cassandra/src/test/java/com/exactpro/cradle/cassandra/dao/messages/MessagesIteratorProviderTest.java (hunk 1 above)
Comment (OptimumCode) [reply cradle-cassandra/src/test/java/com/exactpro/cradle/cassandra/dao/messages/MessagesIteratorProviderTest.java:37]: Okay, got it. Usually, a method like `assertDoesNotThrow` in the assertion framework should be used in such cases. But if this is how it is done in TestNG let's keep it this way

File: cradle-core/src/main/java/com/exactpro/cradle/testevents/StoredTestEventBatch.java [hunk 3]
```diff
@@ -78,7 +79,7 @@ public StoredTestEventBatch(StoredTestEventId id, String name, String type, Stor
 				
 				Set<StoredMessageId> eventMessages = messages != null ? messages.get(child.getId()) : null;
 				if (eventMessages != null)
-					batchMessages.put(child.getId(), Collections.unmodifiableSet(new HashSet<>(eventMessages)));
+					batchMessages.put(child.getId(), Set.copyOf(eventMessages));
```
Comment (Nikita-Smirnov-Exactpro) [inline cradle-core/src/main/java/com/exactpro/cradle/testevents/StoredTestEventBatch.java:82]: Why do we use copy instead of unmodifiableSet ?

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

File: cradle-core/src/main/java/com/exactpro/cradle/testevents/StoredTestEventBatch.java (hunk 3 above)
Comment (lumber1000) [reply cradle-core/src/main/java/com/exactpro/cradle/testevents/StoredTestEventBatch.java:82]: According to code inspection recommendation in IntelliJ IDEA: "Reports java.util.Collections unmodifiable collection calls that can be converted to newer collection factory methods. These can be replaced with e.g. List.of() or Set.of() introduced in Java 9 or List.copyOf() introduced in Java 10."

Comment (Nikita-Smirnov-Exactpro) [reply cradle-core/src/main/java/com/exactpro/cradle/testevents/StoredTestEventBatch.java:82]: These methods are different by complexity
unmodifiableSet - only wrap source collection
copyOf - optionally creates hash set
I think we should expect instance of collection which isn't modified outside of this method.

**Comment by Nikita-Smirnov-Exactpro** [comment]
If we run tests and integration test can we merge this version?

**Pull request merged**